nbformat/validator.py

```diff
@@ -317,7 +318,35 @@ def iter_validate(nbdict=None, ref=None, version=None, version_minor=None,
     if ref:
         errors = validator.iter_errors(nbdict, {'$ref' : '#/definitions/%s' % ref})
     else:
-        errors = validator.iter_errors(nbdict)
+        errors = [e for e in validator.iter_errors(nbdict)]
+
+        if len(errors) > 0 and strip_invalid_metadata:
+            error_tree = validator.error_tree(errors)
+            if "metadata" in error_tree:
+                for key in error_tree["metadata"]:
+                    del nbdict["metadata"][key]
+
+            if "cells" in error_tree:
+                number_of_cells = len(nbdict.get("cells", 0))
+                for cell_idx in range(number_of_cells):
+                    # Cells don't report individual metadata keys as having failed validation
+                    # Instead it reports that it failed to validate against each cell-type definition.
+                    # We have to delve into why those definitions failed to uncover which metadata
+                    # keys are misbehaving.
+                    if "oneOf" in error_tree["cells"][cell_idx].errors:
+                        intended_cell_type = nbdict["cells"][cell_idx]["cell_type"]
+                        schemas_by_index = [ref["$ref"] for ref in error_tree["cells"][cell_idx].errors["oneOf"].schema["oneOf"]]
+                        schema_index = schemas_by_index.index(f"#/definitions/{intended_cell_type}_cell")
+                        for error in error_tree["cells"][cell_idx].errors["oneOf"].context:
+                            rel_path = error.relative_path
+                            error_for_intended_schema = error.schema_path[0] == schema_index
+                            is_top_level_metadata_key = len(rel_path) == 2 and rel_path[0] == "metadata"
+                            if error_for_intended_schema and is_top_level_metadata_key:
+                                nbdict["cells"][cell_idx][rel_path[0]].pop(rel_path[1], None)
```

Comment:
```suggestion
                                nbdict["cells"][cell_idx]["metadata"].pop(rel_path[1], None)
```
maybe less indirection?